### ISSUE TITLE
Cards / Acts issues

### ISSUE THREAD

**Issue by dexington** [dimkanovikov/KITScenarist#957]
I tried using the Cards interface to plan out my scenes. However, after entering about 40 of them I saw that some cards were missing. Then it stopped allowing me to add more cards. I clicked on the many and looked at the script page (no editing) and went back to cards. Then my acts had turned into cards.

I seem to be unable to edit acts in the script. How do you manage acts / scenes in the script so it will show up correctly in cards? 

This seems very buggy.

V. 0.7.2 rc 14 on Linux Ubuntu 20.1

**Comment by david-wendelken** [comment]
Agreed.   When dragging a scenario from one act to another, sometimes it:

1) overwrites the act heading it was just assigned to, destroying that act (but leaving the end of act marker in the script, OR
2) moves things correctly except the name of the scene loses or gains a letter at the front or back of its name/heading.  Ex: SCENE 2.1 might become CENE 2.1, OR
3) it might work correctly.  

In addition, the script order and the card order won't necessarily agree after this happens.   Sometimes stuff disappears from the card view but it's still in the script view.

**Comment by david-wendelken** [comment]
I've also noticed that the act and end of act tags in the script get all tangled up and interleaved when scenarios are moved around.   I suggest naming the acts ACT ONE, ACT TWO and the scenes SCENE 1.1, SCENE 1.2,  SCENE 2.1, etc.  Then move them from act to act both via the cards screen or the script screen.   Things tend to get confused pretty quickly.

**Comment by david-wendelken** [comment]
Had the program lose which acts I have and what's in them again.

I would consider making the ACT and END OF ACT entries editable types of text just like Scene headings if they can't be made reliable.

**Comment by david-wendelken** [comment]
Ok, just learned this tonight.   If you lose a start or end of act line in the script view, delete the matching start or end of act line.

Then, create a scenario header with the name of the act you want to re-create.   Right Click on the outline entry for that scenario and choose Convert to Folder.   This enables you to create a folder where you want it. 

Of course, nothing will be in it, but you can drag items into it with the outline or card view.  (I don't recommend using the script view at this time, there are problems with that.)

**Comment by dexington** [comment]
Yes, exactly. I just a finished a screenplay with it and I've been really happy with it. IT all became easier once I started getting used to the keyboard shortcuts. 
It's actually a great piece of software.

The only thing that's been bugging me is what the SCENE CHARACTER button is for in the sidebar. I ever used it once.

     On Tuesday, January 11, 2022, 07:03:58 PM EST, david-wendelken ***@***.***> wrote:  
 
 


Ok, just learned this tonight. If you lose a start or end of act line in the script view, delete the matching start or end of act line.

Then, create a scenario header with the name of the act you want to re-create. Right Click on the outline entry for that scenario and choose Convert to Folder. This enables you to create a folder where you want it.

Of course, nothing will be in it, but you can drag items into it with the outline or card view. (I don't recommend using the script view at this time, there are problems with that.)

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by dimkanovikov** [comment]
Hey guys,

Sadly, but yes, the cards view has some issues which we can't stable reproduce. Because of that, they are still present in the app. If you find a way how to reproduce incorrect behavior, please share with us these steps.

**Comment by david-wendelken** [comment]
I realized I may have been unclear in my comments.   I've had problems with the acts and scenes showing up incorrectly in the cards view, but I don't know how to make go wrong in the cards view.

I can make them go wrong at will in the script view, i.e., I edit in script view and then switch to cards view.   The cards view will be messed up, but it's just a symptom of the problem created in the script view.

Some ways I know how to do this in the script view:

1) Go to the last line of a scene, a couple characters from the end of the last scene in an act.  Delete one too many characters.  That will destroy the end of act marker as a functional end of act marker.  
2) Go to the first scene heading in an act, a couple of characters before the start of the scene and backspace one character too many.  This will destroy the start of act header as a start of act header.
3) Drag text from one scene to another, but drag it to the line below the end of the last scene in an act.  This will destroy the end of act header as an end of act header.

In all of these situations, you can tell it's happened because the start or end of act headers lose their special coloration in the script editor.     And the card view will reflect this problem.

**Comment by david-wendelken** [comment]
Were you able to replicate the problem using the info in my prior messages in this thread?

**Comment by dimkanovikov** [comment]
Hello @david-wendelken 

I'm sorry, but we are not plan to make any changes to the KIT Scenarist, since we are fully switched to work on the [Story Architect](https://starc.app) app (aka KIT 2.0). You can try to work with it. It already has much more stable cards module.
